node/resolvers/profile/index.ts

```diff
@@ -44,6 +52,38 @@ export const mutations = {
 export const queries = {
   profile: (_: any, { customFields }: any, context: Context) =>
     getProfile(context, customFields),
+
+  checkProfileAllowed: async (_: any, __: any, context: Context) => {
+    const {
+      clients: { catalog, session },
+      vtex: { segment },
+      cookies,
+    } = context
+    const salesChannel = segment ? segment.channel : null
+
+    const { sessionData } = await session.getSession(
+      cookies.get('vtex_session')!,
+      ['*']
+    )
+
+    const email: string | undefined = path(
```

Comment:
no need to do this for typing, you can just: `path<string>(...)`

Reply:
:notbad: